galsim/bandpass.py

```diff
@@ -83,90 +83,122 @@ class Bandpass(object):
                         if throughput is not a LookupTable or file.  See above.]
     @param wave_type    The units to use for the wavelength argument of the `throughput`
                         function. See above for details. [default: 'nm']
+    @param zeropoint    Set the zero-point for this Bandpass.  Here, this can only be a float
+                        value.  See the method `withZeroPoint` for other options for how to
+                        set this using a particular spectrum (AB, Vega, etc.) [default: None]
     """
     def __init__(self, throughput, blue_limit=None, red_limit=None, wave_type='nm',
-                 _wave_list=None):
+                 zeropoint=None, _wave_list=None, _tp=None):
         # Note that `_wave_list` acts as a private construction variable that overrides the way that
         # `wave_list` is normally constructed (see `Bandpass.__mul__` below)
 
-        # Figure out input throughput type.
-        tp = throughput  # For brevity within this function
-        if isinstance(tp, basestring):
-            import os
-            if os.path.isfile(tp):
-                tp = galsim.LookupTable(file=tp, interpolant='linear')
-            else:
-                # Evaluate the function somewhere to make sure it is valid before continuing on.
-                if red_limit is not None:
-                    test_wave = red_limit
-                elif blue_limit is not None:
-                    test_wave = blue_limit
-                else:
-                    # If neither `blue_limit` nor `red_limit` is defined, then the Bandpass should
-                    # be able to be evaluated at any wavelength, so check.
-                    test_wave = 700
-                try:
-                    tp = eval('lambda wave : ' + tp)
-                    tp(test_wave)
-                except:
-                    raise ValueError(
-                        "String throughput must either be a valid filename or something that "+
-                        "can eval to a function of wave. Input provided: {0}".format(throughput))
+        self._orig_tp = throughput  # Save this for pickling.
+        self._tp = _tp              # This is orig_tp turned into an actual function
```

Comment:
Probably I am just being stupid, but I found this comment confusing.  ```self._tp``` is will hold the throughput turned into an actual function later on (after you call ```_initialize_tp```), but right now it's just None, isn't it?

This doesn't matter to users but I think this comment could be clarified to make the code easier to read/parse by developers.